tests/reduce.spec.js

```diff
@@ -0,0 +1,30 @@
+import {reduce} from '../src';
+
+/* eslint-disable require-jsdoc */
+const testArray = [1, 2, 3, 4];
+
+describe('reduce', () => {
+  it('should return the sum of all the values starting with 0', () => {
+    function sumOfValues(accumulator, element) {
+      return accumulator + element;
+    };
+
+    const nativeResult = testArray.reduce(sumOfValues, 0);
+    const robDashResult = reduce(testArray, sumOfValues, 0);
+
+    // robDashResult should be the same value as nativeResult
+    expect(robDashResult).toEqual(nativeResult);
+  });
+
+  it('should return the highest value', () => {
+    function highestValue(accumulator, element) {
+      return accumulator || 0 > element ? accumulator : element;
```

Comment:
What does this function do?

Reply:
Test this out in runjs:
```js
const data = [1, 2, 7, 3, 4]

function reduce(array, callback, initial) {
  let accumulator = initial == undefined ? array[0] : initial;
  const startIndex = initial == undefined ? 1 : 0;
  for (let i = startIndex; i < array.length; ++i) {
    accumulator = callback(accumulator, array[i], i, array);
  }
  return accumulator;
};

function highestValue(accumulator, element) {
      return accumulator || 0 > element ? accumulator : element;
    }

data.reduce(highestValue)
reduce(data, highestValue)
```

Reply:
Forgot the parentheses, oops. Should be:
```
function highestValue(accumulator, element) {
  return (accumulator || 0) > element ? accumulator : element;
}
```

Reply:
What I'm wondering is why you need the optional check on `accumulator`? Why can't you just `return accumulator > element ? ...`?

My understanding is that `accumulator` should never be `undefined`.

Reply:
Yeah, that makes sense to me - I pulled that [function from this article](https://medium.com/poka-techblog/simplify-your-javascript-use-map-reduce-and-filter-bd02c593cc2d) that Al linked me a while back